### ISSUE TITLE
Path tail extractor does not protect against directory traversal attacks

### ISSUE THREAD

**Issue by Kl4rry** [actix/actix-web#2001]
## Expected Behavior
When matching against the url tail any path segment that is equal to .. should be ignored

## Current Behavior
When using this code and the url `http://localhost:8080/..%5c/..%5c/test.txt` the decoded path results in `..\\/..\\/test.txt`  
This allows anyone to read the entire filesystem

## Steps to Reproduce (for bugs)
```
async fn index(req: HttpRequest) -> Result<NamedFile> {
    let tail: PathBuf = req.match_info().query("filename").parse()?;
    println!("{:?}", tail);
    Ok(NamedFile::open(Path::new("public/").join(tail))?)
}

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    use actix_web::{web, App, HttpServer};

    HttpServer::new(|| App::new().route("/{filename:.*}", web::get().to(index)))
        .bind("127.0.0.1:8080")?
        .run()
        .await
}
```

## Context
I just copied the example file server from the documentation. 
According to the documentation this should produce a path that is safe to use as a suffix but it is clearly not.
`As a result of these conditions, a PathBuf parsed from request path parameter is safe to interpolate within, or use as a suffix of, a path without additional checks.`

## Your Environment
rustc 1.52.0-nightly (5fa22fe6f 2021-02-14)
actix-web = "3.3.2"
actix-files = "0.5.0"
Windows 10

**Comment by aliemjay** [comment]
The quoted documentation text was only found prior to v1.0! and I haven't found the example code in the docs.

Please use up-to-date docs, and add links for the quoted text and code!

**Comment by aliemjay** [comment]
Oh, I've just noticed it on [the website](https://actix.rs/docs/url-dispatch/)

Now this is a documentation issue. Currently, path tail extractor matches any string and is not intended to prevent path traversal attacks.

This should be fixed by actix/actix-website#230
